File: src/SlimShader/Shader/TessellatorOutputPrimitive.cs
using SlimShader.Util;

namespace SlimShader.Shader
{
	public enum TessellatorOutputPrimitive
	{
		Undefined = 0,

		[Description("output_point")]
		Point = 1,

		[Description("output_line")]
		Line = 2,

		TriangleCw = 3,
		TriangleCcw = 4
	}
}

File: src/cpp/Source/Chunks/Common/TessellatorOutputPrimitive.cpp
#include "PCH.h"
#include "TessellatorOutputPrimitive.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToStringShex(TessellatorOutputPrimitive value)
{
	switch (value)
	{
	case TessellatorOutputPrimitive::Point :
		return "output_point";
	case TessellatorOutputPrimitive::Line :
		return "output_line";
	case TessellatorOutputPrimitive::TriangleCw :
		return "output_triangle_cw";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}

string SlimShader::ToStringStat(TessellatorOutputPrimitive value)
{
	switch (value)
	{
	case TessellatorOutputPrimitive::Point :
		return "Point";
	case TessellatorOutputPrimitive::Line :
		return "Line";
	case TessellatorOutputPrimitive::TriangleCw :
		return "Clockwise Triangles";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}